File: problems/0020/solution.cpp
#include <iostream>
#include "input.h"
#include "vector"
#include "math.h"
#include "../../helpers/functions.cpp"
using namespace std;

int main()
{
    long sum = 0;
    long num = 2;
    vector<long> originNumberDigits;

    originNumberDigits.push_back(1);

    while (num <= NUMBER)
    {
        long previousExtra = 0;
        vector<long> numberDigits;

        for (vector<long>::iterator it = originNumberDigits.begin(); it != originNumberDigits.end(); ++it)
        {
            long digitCalculation = *it * num + previousExtra;
            long remainder = digitCalculation % 10;
            previousExtra = digitCalculation / 10;
            numberDigits.push_back(remainder);
        }

        while (previousExtra > 0)
        {
            numberDigits.push_back(previousExtra % 10);
            previousExtra /= 10;
        }

        num += 1;
        originNumberDigits = numberDigits;
    }

    for (vector<long>::iterator it = originNumberDigits.begin(); it != originNumberDigits.end(); ++it)
    {
        sum += *it;
    }

    cout << sum << endl;

    return 0;
}

File: problems/0020/solution.js
module.exports = (number) => {
    if (!Number.isInteger(number)) return 0

    let sum = 0
    let num = 2
    let originNumberDigits = [1]

    while (num <= number) {
        let previousExtra = 0
        let numberDigits = []

        originNumberDigits.map((digit) => {
            const digitCalculation = digit * num + previousExtra
            const remainder = digitCalculation % 10
            previousExtra = Math.floor(digitCalculation / 10)
            numberDigits.push(remainder)
        })

        while (previousExtra > 0) {
            numberDigits.push(previousExtra % 10)
            previousExtra = Math.floor(previousExtra / 10)
        }

        num += 1
        originNumberDigits = numberDigits
    }

    originNumberDigits.map((digit) => {
        sum += digit
    })

    return sum
}
